comp/metadata/packagesigning/packagesigningimpl/packagesigning.go

```diff
@@ -0,0 +1,146 @@
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/comp/metadata/internal/util"
+	"github.com/DataDog/datadog-agent/comp/metadata/packagesigning"
+	pkgUtils "github.com/DataDog/datadog-agent/comp/metadata/packagesigning/utils"
+	"github.com/DataDog/datadog-agent/comp/metadata/runner/runnerimpl"
+	"github.com/DataDog/datadog-agent/pkg/serializer"
+	"github.com/DataDog/datadog-agent/pkg/serializer/marshaler"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/DataDog/datadog-agent/pkg/util/hostname"
+	httputils "github.com/DataDog/datadog-agent/pkg/util/http"
+	"go.uber.org/fx"
+)
+
+// Module defines the fx options for this component.
+func Module() fxutil.Module {
+	return fxutil.Component(
+		fx.Provide(newPackageSigningProvider))
+}
+
+const defaultCollectInterval = 12 * time.Hour
+
+// Payload handles the JSON unmarshalling of the metadata payload
+type Payload struct {
+	Hostname  string           `json:"hostname"`
+	Timestamp int64            `json:"timestamp"`
+	Metadata  *signingMetadata `json:"signing_metadata"`
+}
+
+// MarshalJSON serialization a Payload to JSON
+func (p *Payload) MarshalJSON() ([]byte, error) {
+	type PayloadAlias Payload
+	return json.Marshal((*PayloadAlias)(p))
+}
+
+// SplitPayload implements marshaler.AbstractMarshaler#SplitPayload.
+// TODO implement the split
+func (p *Payload) SplitPayload(_ int) ([]marshaler.AbstractMarshaler, error) {
+	return nil, fmt.Errorf("could not split packagesigning payload any more, payload is too big for intake")
+}
+
+type signingMetadata struct {
+	SigningKeys []signingKey `json:"signing_keys"`
+}
+
+type pkgSigning struct {
+	util.InventoryPayload
+
+	log        log.Component
+	conf       config.Component
+	hostname   string
+	pkgManager string
+}
+
+type dependencies struct {
+	fx.In
+
+	Log        log.Component
+	Config     config.Component
+	Serializer serializer.MetricSerializer
+}
+
+type provides struct {
+	fx.Out
+
+	Comp          packagesigning.Component
+	Provider      runnerimpl.Provider
+	FlareProvider flaretypes.Provider
+}
+
+func newPackageSigningProvider(deps dependencies) provides {
+	hname, _ := hostname.Get(context.Background())
+	is := &pkgSigning{
+		conf:       deps.Config,
+		log:        deps.Log,
+		hostname:   hname,
+		pkgManager: getPkgManager(),
+	}
+	is.InventoryPayload = util.CreateInventoryPayload(deps.Config, deps.Log, deps.Serializer, is.getPayload, "signing.json")
+	is.InventoryPayload.MaxInterval = defaultCollectInterval
+	is.InventoryPayload.MinInterval = defaultCollectInterval
+	provider := runnerimpl.NewEmptyProvider()
+	if runtime.GOOS == "linux" {
+		if getPkgManager() != "" {
+			// Package signing telemetry is only valid on Linux and DEB/RPM based distros (including SUSE)
+			provider = is.MetadataProvider()
+		} else {
+			is.log.Info("No supported package manager detected, package signing telemetry will not be collected")
```

Comment:
It would be helpful to add the package manager name as part of this log line. That way, if a customer gets to this condition, the logline would provide enough information for them to come to us.

Reply:
The package manager detection only checks for some files in the host, so I don't have the package manager name. I can only say it's not part of `apt`, `dnf`, `yum` or `zypper`

Reply:
We could add the supported package manager in the log line